### DISCUSSION TITLE
How schedule individual Playbooks / Task Templates per run?

### DISCUSSION THREAD

**Discussion by jrglynn2** [semaphoreui/semaphore#1254] [Q&A]
I have created some Task Templates for playbooks that are used regularly, and can run them ad-hoc as needed.

How do I schedule them? I see that when I create a Task Template there is a Cron field, but I don't believe that does what I want to do...

For example, I have a Task Template that is called "Restart Services", that does exactly what you would expect - restarts the services on a server.

I want to be able to run that Task Template, but schedule it for a specific time (off-hours for that particular server). meaning, the schedule may be different every time I run it - or there will be times I don't need to schedule it, it just runs immediately.

I don't see the option for this, unless it is part of the "Advanced" piece, where I can put in CLI arguments - but if so, I haven't found any examples on how to do that.

Anyone have any ideas?

Thanks!

**Comment by ansibleguy** [comment]
Greetings!

Just to clarify:

* You don't want to run the task repetitive at the same time/in the same interval
* You want to set a one-time auto-execution date-time
  * After this one execution is done - the task is back to only run manually (until you set the next auto-execution date-time)

I guess you could set a cron-time that would only match one time in the year.
Per example: [first of august at 00:00 (0 0 1 8 *)](https://crontab.guru/#0_0_1_8_*)

**Reply by jrglynn2** [reply]
I want to be able to click "Run" on a Task Template, and as one of the options there, choose to schedule that particular run for a certain date & time.

So every time I "Run" a Task Template, I can choose to run it immediately, or scheduled.

Thanks!
